packages/nodes-base/nodes/Kafka/KafkaTrigger.node.ts

```diff
@@ -204,7 +204,7 @@ export class KafkaTrigger implements INodeType {
 
 		const consumer = kafka.consumer({
 			groupId,
-			maxInFlightRequests: this.getNodeParameter('options.maxInFlightRequests', 0) as number,
+			maxInFlightRequests: this.getNodeParameter('options.maxInFlightRequests', null) as number,
```

Comment:
If you can update this to...

`maxInFlightRequests,`

Then between 203 and 205 pop in...

```
const maxInFlightRequests = (
			this.getNodeParameter('options.maxInFlightRequests', null) === 0
				? null
				: this.getNodeParameter('options.maxInFlightRequests', null)
		) as number;
```
This will make sure that even if the Option is set to 0 we swap it to null, To then finish up on line 117 for the `maxInFlightRequests` default set the value to 1 and we should be all good.

Reply:
updated